File: engine/engine-model/src/main/scala/tools/packet/shop/ConfirmShopTransaction.scala
package tools.packet.shop

import constants.ShopTransactionOperation
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ConfirmShopTransaction(private var _operation: ShopTransactionOperation) extends PacketInput {
  def operation: ShopTransactionOperation = _operation

  override def opcode(): SendOpcode = SendOpcode.CONFIRM_SHOP_TRANSACTION
}

File: engine/engine-model/src/main/java/tools/packet/shop/ConfirmShopTransaction.java
package tools.packet.shop;

import constants.ShopTransactionOperation;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ConfirmShopTransaction(ShopTransactionOperation operation) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CONFIRM_SHOP_TRANSACTION;
   }
}